### ISSUE TITLE
userDetailsService.adoc->MyUserDetailsService.groovy fails for RoleGroups implimentations

### ISSUE THREAD

**Issue by coleman-rik** [grails-plugins/grails-spring-security-core#419]
grails-spring-security-core/src/docs/userDetailsService.adoc

MyUserDetailsService.groovy is incorrect if the user is utilizing the new RoleGroup (or AuthorityGroup) methodology.

This section:
```
def authorities = user.authorities.collect {
         new SimpleGrantedAuthority(it.authority)
      }
```
Will fail with a no **authorities** found error.

It need to be updated to:
```
       def conf = SpringSecurityUtils.securityConfig
        String authoritiesPropertyName = conf.userLookup.authoritiesPropertyName
        String authorityPropertyName = conf.authority.nameField

        boolean useGroups = conf.useRoleGroups
        String authorityGroupPropertyName = conf.authority.groupAuthorityNameField

        Person user = Person.findByUsername(username)
        if (!user) throw new NoStackUsernameNotFoundException()

        Collection<?> userAuthorities = user."$authoritiesPropertyName"
        def authorities
        if (useGroups) {
            if (authorityGroupPropertyName) {
                authorities = userAuthorities.collect { it."$authorityGroupPropertyName" }.flatten().unique().collect { new SimpleGrantedAuthority(it."$authorityPropertyName") }
            }
            else {
                log.warn "Attempted to use group authorities, but the authority name field for the group class has not been defined."
            }
        }
        else {
            authorities = userAuthorities.collect { new SimpleGrantedAuthority(it."$authorityPropertyName") }
        }
        authorities ?: [NO_ROLES]
```

**Comment by burtbeckwith** [comment]
Fixed in v3.0.4

**Comment by coleman-rik** [comment]
Any chance that things in the 2.x branch will get updated?
